pkg/util/ipset/ipset.go

```diff
@@ -92,35 +92,33 @@ type IPSet struct {
 }
 
 // Validate checks if a given ipset is valid or not.
-func (set *IPSet) Validate() bool {
+func (set *IPSet) Validate() error {
```

Comment:
Slightly larger comment:  This API is very weak on "hiding".  

I wonder if it would be better to make a bigger change and use a constructor, like:

```
type ArgFunc func(s *ipSet) error

func New(name string, type Type, comment string, args ...ArgFunc) Interface {
    //validate params, return error if needed
    s := ipSet {
        name: name,
        type: type,
        comment: comment,
    }
    // call each fn(&s) in args, return error if needed
    return s
}

func HashArgs(family string, hashSize,  maxElem int) ArgFunc {
     return func(s *ipSet) error {
        // validate args and return error if needed
        s.family = family
        s.hashSize = hashSize
        s.maxElem = maxElem
        return nil
    }
}

func PortArgs(range string) ArgFunc {
    return func(s *ipSet) error {
        // validate args and return error if needed
        s.range = range
    }
}
```

or even this (I am not an ipset expert so my name choices may not be good :):

```
func NewHashSet(name string, type Type, comment string, family string, hashSize,  maxElem int) Interface {...}
func NewPortSet(name string, type Type, comment string, range string) Interface {...}
```

Also HashFamily could be a type with const values, like `Type`

The point being to hide as much detail as possible and export as few things as possible.  This lib isn't used a ton, so fixing clients should be OK.

This could be a followup, of course.

Reply:
Agree, sounds like a big change, and good follow up for sure

Reply:
We should create an issue for this conversation

Reply:
https://github.com/kubernetes/kubernetes/issues/109034